kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/PostCommitSnapshotSuite.scala

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.defaults
+
+import scala.collection.immutable.Seq
+
+import io.delta.kernel.{Operation, Snapshot, TransactionCommitResult}
+import io.delta.kernel.defaults.utils.{TestRow, WriteUtilsWithV2Builders}
+import io.delta.kernel.engine.Engine
+import io.delta.kernel.expressions.Literal
+import io.delta.kernel.internal.{InternalScanFileUtils, SnapshotImpl, TableConfig}
+import io.delta.kernel.test.MockEngineUtils
+import io.delta.kernel.types.IntegerType.INTEGER
+import io.delta.kernel.utils.CloseableIterable.inMemoryIterable
+
+import org.scalatest.funsuite.AnyFunSuite
+
+/**
+ * Test suite for validating the behavior of postCommitSnapshot in various scenarios.
+ *
+ * Note that we use "PCS" in our test names for brevity.
+ */
+class PostCommitSnapshotSuite
+    extends AnyFunSuite
+    with WriteUtilsWithV2Builders
+    with MockEngineUtils {
+
+  //////////////////
+  // Test Helpers //
+  //////////////////
+
+  private def assertAddFilesMatch(
+      engine: Engine,
+      actual: SnapshotImpl,
+      expected: SnapshotImpl): Unit = {
+    val actualFiles = collectScanFileRows(actual.getScanBuilder.build(), engine)
+      .map(x => InternalScanFileUtils.getAddFileStatus(x).getPath)
+    val expectedFiles = collectScanFileRows(expected.getScanBuilder.build(), engine)
+      .map(x => InternalScanFileUtils.getAddFileStatus(x).getPath)
+
+    assert(actualFiles === expectedFiles)
+  }
+
+  private def checkPostCommitSnapshot(
+      engine: Engine,
+      postCommitSnapshot: Snapshot,
+      expectCrc: Boolean = false): Unit = {
+    val actual = postCommitSnapshot.asInstanceOf[SnapshotImpl]
+    val expected = latestSnapshot(actual.getPath, engine)
+
+    if (expectCrc) {
+      assert(actual.getCurrentCrcInfo.isPresent)
+    }
+
+    // TODO: We need better visibility into when the below information is loaded from the log,
+    //       loaded from CRC, or already stored in memory (i.e. injected during post-commit snapshot
+    //       creation)
```

Comment:
We do test this stuff in the EngineMetricsSuites I think btw if any of that is useful

Reply:
Thanks! I think we can revisit this later. For now, we are at least verifying that the post commit snapshot is able to return and expose the right information. Plus, I have a test below explicitly for ICT.